orttraining/orttraining/eager/opgen/opgen/generator.py

```diff
@@ -424,6 +460,20 @@ def _write_function_body(self, writer: writer.SourceWriter, mapped_func: MappedO
                                     in_place_params[0] = cpp_param.identifier.value
                                     break
 
+                # if no in_place_params found and there is an out input to set
+                # and this is the last onnx op, we set the out to be written to
+                if len(in_place_params) == 0 and set_out_tensor and onnx_op_index == (len(ctx.ops) - 1):
+                    if cast_output_to_bool:
+                        # if the out type is bool already we can still set it since no cast will be needed.
+                        writer.writeline("if (out.scalar_type() == at::kBool) {")
```

Comment:
do you need a "else" branch in the generated code?

Reply:
The else branch is the primary branch, as it sets the "out" in all cases but Equal. If I don't do that, I'll have to do a copy (`copy(invoker, {onnx_op.outputs}[0], ort_input_out)`) which I am trying to avoid.

Reply:
I have since removed this since I was able to put the CAST in the atenops.py config instead.